src/main/java/edu/iu/terracotta/service/app/impl/AssessmentServiceImpl.java

```diff
@@ -93,6 +109,20 @@ public AssessmentDto postAssessment(AssessmentDto assessmentDto, long treatmentI
         return toDto(assessmentSaved, false, false, false, false);
     }
 
+    private AssessmentDto toDto(Assessment assessment, Participant participant, boolean submissions) throws AssessmentNotMatchingException {
+        AssessmentDto assessmentDto = toDto(assessment, null, false, false, submissions, false);
```

Comment:
So when `submissions` is true, this will return submissions from all participants for the assignment. Maybe this method could pass false for submissions and then just load the participant's submissions here. Also, only dateSubmitted != null submissions should be included.

Reply:
I'm on board with that! I'll fix this up